src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/detailsAugmentation.hpp

```diff
@@ -29,6 +29,32 @@ class TDetailsAugmentation final : public AbstractHandler<std::shared_ptr<TScanC
 {
 private:
     std::shared_ptr<TDatabaseFeedManager> m_databaseFeedManager;
+    const std::string ECS_VERSION = "8.11.0";
+
+    /**
+     * @brief Populates a field in a JSON object with the provided value, if it's not empty (for string values).
+     *
+     * This function checks if the provided value is a non-empty string and populates the specified JSON field with it.
+     * If the value is an empty string, it returns without making any changes to the JSON object.
+     *
+     * @tparam T The type of the value to populate the JSON field with.
+     * @param json The JSON object to populate.
+     * @param key The JSON pointer specifying the field.
+     * @param value The value to populate the field with (as std::string_view).
+     */
+    template<typename T>
+    void populateField(nlohmann::json& json, const nlohmann::json::json_pointer& key, const std::string_view value)
```

Comment:
not all values should be a string, the json values could be numbers, floats, null, and... strings.

Reply:
probably we can use:

```cpp
template<typename T>
void populateField(nlohmann::json& json, const nlohmann::json::json_pointer& key, T&& value)
{
    if constexpr (std::is_same_v<std::remove_cv_t<std::remove_reference_t<T>>, std::string_view> || std::is_same_v<T, std::string>)
    {
        if (!value.empty())
        {
            json[key] = value;
        }
    }
    else
    {
        json[key] = value;
    }
}
```

Reply:
Done!